### DISCUSSION TITLE
Make file tags searchable

### DISCUSSION THREAD

**Discussion by nickluger** [directus/directus#6893] [Feature Requests]
[Directus files](https://docs.directus.io/reference/api/system/files/#the-file-object) have a `tags` field. 

Tagging files - especially images - is great.

The thing is: Tags are **not searchable** via the search field and images are not filterable on selection as described [in this discussion](https://github.com/directus/directus/discussions/6097). So tags become basically useless in the app. (They might be handy via API of course)

So either tags should be searchable via the search box, or the _Advanced Filter_ should be added to the serach bar, as proposed by @benhaynes [here](https://github.com/directus/directus/discussions/6097#discussioncomment-835038).

I opened a [separate discussion](https://github.com/directus/directus/discussions/6892) for that second option.

**Comment by rijkvanzanten** [comment]
Tags are saved as a CSV string in the database, making a robust search a little more difficult. We could consider making tags a o2m/m2m 🤔  (you could always add a custom field to directus_files that is relational field to a new `tags` collection)

**Reply by gmarton** [reply]
In a data model I use a normal input (string) field - named it "Tags" to store the same CSV (or even JSON like: [ "cat", "dog" ]) and search works that way, just wont have useful tagging feature that the Tags field has. 
It would be nice to have the search include the field in the same database table added with the Tags. I searched the issues/discussions but most would want to search relational data stored in different tables like #10517

Tried adding o2m/m2m tags collection as @rijkvanzanten suggested, filters do work that way just not the search.
I see that REST API returns a string for input field and array for tags field even tough they are stored the same in the db.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
